### PR TITLE
Ensure we can create kademlia with custom storage

### PR DESCRIPTION
**Pull request by egonelbre**
Necessary for testplanet

### CODE REVIEW

File: pkg/kademlia/kademlia.go [hunk 1]
```diff
@@ -70,16 +70,20 @@ func NewKademlia(id dht.NodeID, bootstrapNodes []pb.Node, address string, identi
 	}
 	kdb, ndb := dbs[0], dbs[1]
 
-	rt, err := NewRoutingTable(&self, kdb, ndb, &RoutingOptions{
-		idLength:     kadconfig.DefaultIDLength,
-		bucketSize:   kadconfig.DefaultBucketSize,
-		rcBucketSize: kadconfig.DefaultReplacementCacheSize,
+	rt, err := NewRoutingTable(self, kdb, ndb, &RoutingOptions{
+		IDLength:     kadconfig.DefaultIDLength,
```
Comment (coyle) [inline pkg/kademlia/kademlia.go]: Why are we exporting these variables? They should be changed by anything external.

File: pkg/kademlia/kademlia.go [hunk 2]
```diff
@@ -70,16 +70,20 @@ func NewKademlia(id dht.NodeID, bootstrapNodes []pb.Node, address string, identi
...
-
 	if err != nil {
 		return nil, BootstrapErr.Wrap(err)
 	}
 
+	return NewKademliaWithRoutingTable(self, bootstrapNodes, identity, kadconfig.Alpha, rt)
+}
+
+// NewKademliaWithRoutingTable returns a newly configured Kademlia instance
+func NewKademliaWithRoutingTable(self pb.Node, bootstrapNodes []pb.Node, identity *provider.FullIdentity, alpha int, rt *RoutingTable) (*Kademlia, error) {
```
Comment (coyle) [inline pkg/kademlia/kademlia.go]: Why not just `NewKademlia`? every kademlia should have a routing table

Comment (jtolds) [reply pkg/kademlia/kademlia.go]: this is probably not relevant, but you can have a service be a kademlia client without a routing table. if all you're doing is lookups you can just kick off some workers against some well known nodes you have.

File: pkg/kademlia/kademlia.go (hunk 1 above)
Comment (egonelbre) [reply pkg/kademlia/kademlia.go]: We need a way to create routing table with custom database outside from this package. (Starting an inmemory instance of the whole thing)

File: pkg/kademlia/kademlia.go (hunk 2 above)
Comment (egonelbre) [reply pkg/kademlia/kademlia.go]: This is just so I can provide a RoutingTable initialized with a inmemory database, without disrupting the existing functions.

File: pkg/kademlia/kademlia.go (hunk 1 above)
Comment (egonelbre) [reply pkg/kademlia/kademlia.go]: I would also be fine with just having constants and not being able to customize these things. Or having a default routing options.

**Pull request merged**